manifests/config.pp

```diff
@@ -190,10 +190,17 @@
   }
 
   if $use_config_file_for_plugins {
+    $managent_plugin = $admin_enable or $management_enable ? { true => 'rabbitmq_management', false => undef }
+    $stomp_plugin = $stomp_ensure ? { true => 'rabbitmq_stomp', false => undef }
+    $auth_backend_ldap_plugin = $ldap_auth ? { true => 'rabbitmq_auth_backend_ldap', false => undef }
+    $shovel_plugin = $config_shovel ? { true => 'rabbitmq_shovel', false => undef }
+    $shovel_management_plugin = $config_shovel and ($admin_enable or $management_enable) ? { true => 'rabbitmq_shovel_management', false => undef }
+
+    $_plugins = join($plugins, $managent_plugin, $stomp_plugin, $auth_backend_ldap_plugin, $shovel_plugin, $shovel_management_plugin)
```

Comment:
I don't think this will work
https://www.puppet.com/docs/puppet/7/function.html#join
```
     Puppet::PreformattedError:
       Evaluation Error: Error while evaluating a Function Call, 'join' expects between 1 and 2 arguments, got 6
```
`join()` in puppet expects an array and a delimiter.

I think you can populate an array with all those values another way, but even then, you may get an issue because `$_plugins` may not be accessible from the template, whether via `$_plugins` or `$::rabbitmq::_plugins`?

Reply:
```
    $_plugins = [$plugins + $managent_plugin + $stomp_plugin + $auth_backend_ldap_plugin + $shovel_plugin + $shovel_management_plugin]
```
Maybe not the most idiomatic, but maybe this gets closer to what you're trying?